extensions/wikia/NjordPrototype/scripts/Mom.js

```diff
@@ -0,0 +1,303 @@
+		}, addEmpty = function () {
+			var $new = $(document.createElement('div')),
+				$button = $(document.createElement('div'));
+			$button.text('ADD');
+			$button.addClass('add-btn new-btn');
+			$new.addClass('mom-module mom-add-module');
+			var $secButton = $button.clone(true),
+				$secNew = $new.clone();
+			$button.on('click', {align: 'right'}, addNewModule);
+			$secButton.on('click', {align: 'left'}, addNewModule);
+
+			$rightColumn.append($new.append($button));
+			$leftColumn.append($secNew.append($secButton));
+		}, removeEmpty = function () {
+			$('.mom-add-module').remove();
+		}, addNewModule = function (ev) {
+			$mainContentContainer.startThrobbing();
+			//add placeholder will be replace after loaded
+			$('<div id="MomNewPlaceHolder" class="mom-no-display"></div>').insertBefore($(this).parent());
+			$.nirvana.sendRequest({
+				controller: 'NjordController',
+				method: 'modula',
+				format: 'HTML',
+				type: 'GET',
+				data: {
+					align: ev.data.align,
+					title: 'new'
+				},
+				callback: onAddNewBlock,
+				onErrorCallback: function () {
+					$mainContentContainer.stopThrobbing();
+					// TODO: handle failure
+				}
+			});
+		}, onAddNewBlock = function (d) {
+			var $this = $(d);
+			$('#MomNewPlaceHolder').replaceWith($this);
+			refresh();
+			$this.addClass('mom-hidden');
+			$this.find('.mom-bar').css('display', 'flex');
+			$this.find('.mom-overlay').show();
+			$this.find('.mom-bar-content').attr('contenteditable', true);
+			$this.find('.mom-delete-btn').on('click', function () {
+				$(this).parents('.mom-module').remove();
+			});
+			$mainContentContainer.stopThrobbing();
+		}, refresh = function () {
+			$deleteButton.refresh();
+			$momOverlays.refresh();
+			$momBar.refresh();
+			$momBarContent.refresh();
+			$moms.refresh();
+			$moduleEditButton.refresh();
+			$moduleDiscardButton.refresh();
+			$moduleSaveButton.refresh();
+		};
+
+	if (window.wgUserName) {
+		initMom();
+	}
+	$mwContent.on('click', '.mom-edit-btn', function () {
+		var $moduleContainer = $(this).parents('.mom-module');
+		$moduleContainer.startThrobbing();
+
+		$.nirvana.sendRequest({
+			controller: 'NjordController',
+			method: 'getWikiMarkup',
+			type: 'GET',
+			data: {
+				articleTitle: $moduleContainer.attr('data-title')
+			},
+			callback: function (resp) {
+				$moduleContainer.removeClass('mom-hidden');
+				$moduleContainer.addClass('mom-edit');
+				$('.mom-save-btn', $moduleContainer).show();
+				$('.mom-discard-btn', $moduleContainer).show();
+
+				$('.mom-edit-btn', $moduleContainer).hide();
+				$('.mom-delete-btn', $moduleContainer).hide();
+				$('.mom-content', $moduleContainer).hide();
+				$('.mom-overlay', $moduleContainer).hide();
+
+				$('.mom-wiki-markup .wiki-markup', $moduleContainer).val(resp['wikiMarkup']);
+				$('.mom-wiki-markup', $moduleContainer).show();
+				$moduleContainer.stopThrobbing();
+			},
+			onErrorCallback: function () {
+				// TODO: handle failure
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)